apps/server/src/modules/files-storage/controller/files-storage.consumer.spec.ts

```diff
@@ -0,0 +1,157 @@
+		jest.clearAllMocks();
+	});
+
+	afterAll(async () => {
+		await orm.close();
+		await module.close();
+	});
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+		module = await Test.createTestingModule({
+			providers: [
+				FilesStorageConsumer,
+				{
+					provide: FileRecordUC,
+					useValue: createMock<FileRecordUC>(),
+				},
+				{
+					provide: FilesStorageUC,
+					useValue: createMock<FilesStorageUC>(),
+				},
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+				{
+					provide: MikroORM,
+					useValue: orm,
+				},
+			],
+		}).compile();
+
+		fileRecordUC = module.get(FileRecordUC);
+		filesStorageService = module.get(FilesStorageUC);
+		service = module.get(FilesStorageConsumer);
+	});
+
+	describe('copyFilesOfParent()', () => {
+		const schoolId: EntityId = new ObjectId().toHexString();
+		it('should call filesStorageService.copyFilesOfParent with params', async () => {
+			const payload = {
+				userId: new ObjectId().toHexString(),
+				source: {
+					parentId: new ObjectId().toHexString(),
+					parentType: FileRecordParentType.Course,
+					schoolId,
+				},
+				target: {
+					parentId: new ObjectId().toHexString(),
+					parentType: FileRecordParentType.Course,
+					schoolId,
+				},
+			};
+			await service.copyFilesOfParent(payload);
+			expect(filesStorageService.copyFilesOfParent).toBeCalledWith(payload.userId, payload.source, {
+				target: payload.target,
+			});
+		});
+
+		it('regular RPC handler should receive a valid RPC response', async () => {
+			const sourceCourse = courseFactory.buildWithId();
+			const targetCourse = courseFactory.buildWithId();
+			const payload = {
+				userId: new ObjectId().toHexString(),
+				source: {
+					parentId: sourceCourse.id,
+					parentType: FileRecordParentType.Course,
+					schoolId: sourceCourse.school.id,
+				},
+				target: {
+					parentId: targetCourse.id,
+					parentType: FileRecordParentType.Course,
+					schoolId: targetCourse.school.id,
+				},
+			};
+			filesStorageService.copyFilesOfParent.mockResolvedValue([[], 0]);
+			const response = await service.copyFilesOfParent(payload);
+			expect(response.message).toEqual([]);
+		});
+	});
+
+	describe('fileRecordsOfParent()', () => {
+		const schoolId: EntityId = new ObjectId().toHexString();
+		it('should call filesStorageService.fileRecordsOfParent with params', async () => {
+			const payload = {
+				parentId: new ObjectId().toHexString(),
+				parentType: FileRecordParentType.Course,
+				schoolId,
+			};
+			fileRecordUC.fileRecordsOfParent.mockResolvedValue([[], 0]);
+			await service.fileRecordsOfParent(payload);
+			expect(fileRecordUC.fileRecordsOfParent).toBeCalledWith('REMOVE_IF_MOVED_TO_SERVICE', payload);
+		});
+
+		it('should return array instances of FileRecordResponse', async () => {
+			const payload = {
+				parentId: new ObjectId().toHexString(),
+				parentType: FileRecordParentType.Course,
+				schoolId: new ObjectId().toHexString(),
```

Comment:
One test use schoolId from top, the other create a new id? 
It is unclear for me what is happend in this tests from descriptions. 
The one have empty result the other the fileRecords ok, but way not prepared this both situations more transparent. 

The test no file found and the test parent type do not exists are different.